runtime/runtime-params-estimator/estimator-warehouse/src/check.rs

```diff
@@ -92,34 +68,227 @@ pub(crate) fn check(db: &Db, config: &CheckConfig) -> anyhow::Result<()> {
+        }
+        _ => anyhow::bail!("You have to either specify both commits for comparison or neither."),
+    };
+    let estimations = if config.estimations.len() > 0 {
+        config.estimations.clone()
+    } else {
+        let rows_a = EstimationRow::select_by_commit_and_metric(db, &commit_after, config.metric)?;
+        let rows_b = EstimationRow::select_by_commit_and_metric(db, &commit_before, config.metric)?;
+        let estimations_a = rows_a.into_iter().map(|row| row.name).collect::<BTreeSet<_>>();
+        let estimations_b = rows_b.into_iter().map(|row| row.name).collect::<BTreeSet<_>>();
+        estimations_a.intersection(&estimations_b).cloned().collect()
+    };
+    let warnings =
+        estimation_changes(db, &estimations, &commit_before, &commit_after, 0.1, config.metric)?;
+    Ok((commit_before, commit_after, warnings))
+}
+
 fn estimation_changes(
     db: &Db,
     estimation_names: &[String],
-    commit_a: &str,
-    commit_b: &str,
+    commit_before: &str,
+    commit_after: &str,
     tolerance: f64,
     metric: Metric,
 ) -> anyhow::Result<Vec<Notice>> {
     let mut warnings = Vec::new();
     for name in estimation_names {
-        let a = &EstimationRow::get(db, name, commit_a, metric)?[0];
-        let b = &EstimationRow::get(db, name, commit_b, metric)?[0];
-        let rel_change = if a.gas == 0.0 { 100.0 } else { (a.gas - b.gas).abs() / a.gas };
+        let b = &EstimationRow::get(db, name, commit_before, metric)?[0];
+        let a = &EstimationRow::get(db, name, commit_after, metric)?[0];
+        let rel_change = if b.gas == 0.0 { 1.0 } else { (b.gas - a.gas).abs() / b.gas };
         if rel_change > tolerance {
             warnings.push(Notice::RelativeChange(RelativeChange {
                 estimation: name.clone(),
-                before: a.gas,
-                after: b.gas,
+                before: b.gas,
+                after: a.gas,
             }))
         }
     }
 
     Ok(warnings)
 }
+
+#[cfg(test)]
+mod test {
+    use super::{inner_check, CheckConfig, Notice, RelativeChange};
+    use crate::db::{Db, EstimationRow};
+    use crate::Metric;
+
+    #[track_caller]
+    fn check_estimation_changes(
+        db: &Db,
+        estimation_names: impl Iterator<Item = &'static str>,
+        commit_before: &str,
+        commit_after: &str,
+        tolerance: f64,
+        metric: Metric,
+        expected: Vec<Notice>,
+    ) {
+        let output = super::estimation_changes(
+            &db,
+            &estimation_names.map(|s| s.to_owned()).collect::<Vec<_>>(),
+            commit_before,
+            commit_after,
+            tolerance,
+            metric,
+        )
+        .unwrap();
+        assert_eq!(expected, output);
+    }
+
+    #[test]
+    fn test_estimation_changes() {
+        let db = Db::test();
+        // Data with large difference between commits
+        EstimationRow::insert_test_data_set(&db, "LogBase", 1e9, 1e9, Metric::ICount, 3);
```

Comment:
So yeah, this `insert_test_data_set` things looks potentially churn prone, as it is made of code, not of data. Can we use `Db::import_json_lines` here to aviod hard-coding signatures?